### ISSUE TITLE
Dynamic Routing: prev/next helper should take collections into account

### ISSUE THREAD

**Issue by kirrg001** [TryGhost/Ghost#10059]
Reported here: https://forum.ghost.org/t/hiding-content-collection-in-suggested-posts/3716

## Summary

If you define more than one collection, the prev/next helper shows posts from another collection.
The helper should take the collection configuration into account. That basically means, it has to use the filter from the target collection.

## Steps to reproduce

1. Add two collections.
2. Filter by a tag.
3. Use the prev/next helper.
4. Serve a post from a collection. The helper shows posts from the other collection too.

---

I have not looked into a solution yet.

Ghost: master

**Comment by ahawkins** [comment]
I 👍 this issue. Here's my use case. I use a `#newsletter` collection with the RSS feed to power my mailing list. Since I'm the only author on my site, `#newsletter` stuff shows up in the next/previous posts.

**Comment by ahawkins** [comment]
Why was this closed? Will it be fixed?

// Adam

**Comment by ErisDS** [comment]
All issues tagged with `routing` have been tagged with `later` and closed until we prioritise working on the Routing feature again.

**Comment by seyfeddin** [comment]
Any updates on this? Anything we can help on?

**Comment by naz** [comment]
This issue is not something easily fixable and we'd need time to properly design where the "filter" piece of data would fit in template rendering process/state. 

### TLDR
Root of the problem has to do with how routing layer passes data to helpers like prev/next - during render stage helpers don't have enough information to know about collection route's filter.

### Longer version
When the template is rendered in routing rendered [here](https://github.com/TryGhost/Ghost/blob/52bcf6086a71e9d2a008f7f873e5e48a2bec8e75/core/frontend/services/routing/helpers/renderer.js#L29), it's the last place where `res` maintains state about collection route's [filter](https://github.com/TryGhost/Ghost/blob/52bcf6086a71e9d2a008f7f873e5e48a2bec8e75/core/frontend/services/routing/CollectionRouter.js#L102). 

To provide additional data all the way to prev/next helper (and other helpers if needed) we could use `root` property available in `root` state, similar to how [context](https://github.com/TryGhost/Ghost/blob/52bcf6086a71e9d2a008f7f873e5e48a2bec8e75/core/frontend/helpers/prev_post.js#L77) and [apiVersion](https://github.com/TryGhost/Ghost/blob/52bcf6086a71e9d2a008f7f873e5e48a2bec8e75/core/frontend/helpers/prev_post.js#L47) are passed around.

### Note on reproduction steps
To reproduce 2 collection state: 
0. Use default Casper theme
1. Upload following `routes.yaml`:
```yaml
routes:
  /signup/: members/signup
  /signin/: members/signin
  /account/: members/account

collections:
  /one/:
    permalink: /one/{slug}/
    template: index
    filter: 'tag:one'
  /two/:
    permalink: /two/{slug}/
    template: index
    filter: 'tag:two'

taxonomies:
  tag: /tag/{slug}/
  author: /author/{slug}/
```
2. Create multiple posts with tags "one" and "two"
3. Navigate to `http://localhost:2368/one/`
4. Navigate to any of the posts available in the collection
5. Observe a bug on the bottom "prev/next" section

**Comment by dschwertfeger** [comment]
I ran into this too today and Google brought me here. 

What's the best workaround for this until it's fixed? What do you recommend?

**Comment by naz** [comment]
Related issue https://github.com/TryGhost/Themes/issues/246
